Aquifer/Proxy.swift

```diff
@@ -8,90 +8,44 @@
-        case let .Respond(dO, fDI): return ProxyRepr<NO, NI, DI, DO, FR>.Respond(dO) { fDI($0).requestBind(f) }
-        case let .Pure(x): return ProxyRepr<NO, NI, DI, DO, FR>.Pure(x)
-        }
-    }
-
-    internal func pushBind<NI, NO>(f: DO -> ProxyRepr<DI, DO, NI, NO, FR>) -> ProxyRepr<UO, UI, NI, NO, FR> {
-        switch self {
-        case let .Request(uO, fUI): return ProxyRepr<UO, UI, NI, NO, FR>.Request(uO) { pushBindExt(fUI($0), f) }
-        case let .Respond(dO, fDI): return pullBindExt(f(dO()), fDI)
-        case let .Pure(x): return ProxyRepr<UO, UI, NI, NO, FR>.Pure(x)
-        }
-    }
-
-    internal func pullBind<NO, NI>(f: UO -> ProxyRepr<NO, NI, UO, UI, FR>) -> ProxyRepr<NO, NI, DI, DO, FR> {
-        switch self {
-        case let .Request(uO, fUI): return pushBindExt(f(uO()), fUI)
-        case let .Respond(dO, fDI): return ProxyRepr<NO, NI, DI, DO, FR>.Respond(dO) { pullBindExt(fDI($0), f) }
-        case let .Pure(x): return ProxyRepr<NO, NI, DI, DO, FR>.Pure(x)
-        }
-    }
-}
-
-private func pushBindExt<UO, UI, DI, DO, NI, NO, FR>(p: ProxyRepr<UO, UI, DI, DO, FR>, _ f: DO -> ProxyRepr<DI, DO, NI, NO, FR>) -> ProxyRepr<UO, UI, NI, NO, FR> {
-    return p.pushBind(f)
-}
-
-private func pullBindExt<UO, UI, DI, DO, NI, NO, FR>(p: ProxyRepr<UO, UI, DI, DO, FR>, _
-    f: UO -> ProxyRepr<NO, NI, UO, UI, FR>) -> ProxyRepr<NO, NI, DI, DO, FR> {
-        return p.pullBind(f)
-}
-
 /// A bidirectional channel for information.
 ///
+/// A `Proxy` is so named because it can represent many different kinds of information flows.  There
+/// are 6 overarching specific types that a `Proxy` can represent, each with separate semantics.
+///
+///     /// An effectful computation.
+///     ///
+///     /// `Effect`s neither await nor yield.
+///     typealias Effect<Result> = Proxy<X, (), (), X, Result>
+///
+///     /// A computation that yields values of type `B`.
+///     ///
+///     /// `Producer`s can only yield.
+///     typealias Producer<B, Result> = Proxy<X, (), (), B, Result>
+///
+///     /// A computation that can await values of type `A` and yield values of type `B`.
+///     ///
+///     /// `Pipe`s can both await and yield.
+///     typealias Pipe<A, B, Result> = Proxy<(), A, (), B, Result>
+///
+///     /// A computation that can await values of type `A`.
+///     ///
+///     /// `Consumer`s can only await.
+///     typealias Consumer<A, Result> = Proxy<(), A, (), X, Result>
+///
+///
+///     /// Sends requests of type `RequestT` and recieves responses of type `RespondT`.
+///     ///
+///     /// `Client`s only request and never respond.
+///     typealias Client<RequestT, RespondT, Result> = Proxy<RequestT, RespondT, (), X, Result>
+///
+///     /// Receives values of type `ReceiveT` and responds with values of type `RespondT`.
+///     ///
+///     /// `Server`s only respond and never request.
+///     typealias Server<ReceiveT, RespondT, Result> = Proxy<X, (), ReceiveT, RespondT, Result>
```

Comment:
Mhm, but I'm looking for a style we can use in the docs consistently.  For example

```swift
/// Converts a given sequence into a pipe that produces elements of the same.
public func each<UO, UI, S: SequenceType>(seq: S) -> Proxy<UO, UI, (), S.Generator.Element, ()> {
    return Proxy(eachRepr(seq.generate()))
}
```

Can become

```
/// Converts a given sequence into a `Producer` of the elements of the sequence.
public func each<UO, UI, S: SequenceType>(seq: S) -> Proxy<UO, UI, (), S.Generator.Element, ()> {
    return Proxy(eachRepr(seq.generate()))
}
```

or 

```
/// Converts a given sequence into a pipe that produces elements of the same.
///
/// Yields a Producer<S.Generator.Element, ()>
public func each<UO, UI, S: SequenceType>(seq: S) -> Proxy<UO, UI, (), S.Generator.Element, ()> {
    return Proxy(eachRepr(seq.generate()))
}
```